impeller/entity/geometry/round_superellipse_geometry.cc

```diff
@@ -0,0 +1,306 @@
+  // and right squircles come from another two (height-aligned).
+  //
+  // Denote the distance from the center of the square-like squircles to the
+  // origin as `c`. The width-aligned square-like squircle and the
+  // height-aligned one have the same offset in different directions.
+  const Scalar c = (size.width - size.height) / 2;
+
+  // Draw the first quadrant of the shape and store in `points`. It will be
+  // mirrored to other quadrants later.
+  std::vector<Point> points;
+  points.reserve(41);
+
+  DrawOctantSquareLikeSquircle(points, size.width, corner_radius_, Point{0, -c},
+                               false);
+  points.push_back(Point(size / 2) - gap(corner_radius_));  // Point M
+  DrawOctantSquareLikeSquircle(points, size.height, corner_radius_, Point{c, 0},
+                               true);
+
+  static constexpr Point reflection[4] = {{1, 1}, {1, -1}, {-1, -1}, {-1, 1}};
+
+  // Reflect the 1/4 arc into the 4 quadrants and generate the tessellated mesh.
+  // The iteration order is reversed so that the trianges are continuous from
+  // quadrant to quadrant.
+  std::vector<Point> geometry;
+  geometry.reserve(1 + 4 * points.size());
+  geometry.push_back(center);
+  // All arcs include the starting point and exclude the ending point.
+  for (auto i = 0u; i < points.size() - 1; i++) {
+    geometry.push_back(center + (reflection[0] * points[i]));
+  }
+  for (auto i = points.size() - 1; i >= 1; i--) {
+    geometry.push_back(center + (reflection[1] * points[i]));
+  }
+  for (auto i = 0u; i < points.size() - 1; i++) {
+    geometry.push_back(center + (reflection[2] * points[i]));
+  }
+  for (auto i = points.size() - 1; i >= 1; i--) {
+    geometry.push_back(center + (reflection[3] * points[i]));
+  }
+  geometry.push_back(center + points[0]);
+
+  std::vector<uint16_t> indices;
+  indices.reserve(geometry.size() * 3);
+  for (auto i = 2u; i < geometry.size(); i++) {
+    indices.push_back(0);
+    indices.push_back(i - 1);
+    indices.push_back(i);
+  }
+
+  auto& host_buffer = renderer.GetTransientsBuffer();
+  return GeometryResult{
+      .type = PrimitiveType::kTriangle,
+      .vertex_buffer =
+          {
+              .vertex_buffer = host_buffer.Emplace(
+                  geometry.data(), geometry.size() * sizeof(Point),
+                  alignof(Point)),
+              .index_buffer = host_buffer.Emplace(
+                  indices.data(), indices.size() * sizeof(uint16_t),
+                  alignof(uint16_t)),
+              .vertex_count = indices.size(),
+              .index_type = IndexType::k16bit,
+          },
+      .transform = entity.GetShaderTransform(pass),
+  };
+}
+
+std::optional<Rect> RoundSuperellipseGeometry::GetCoverage(
+    const Matrix& transform) const {
+  return bounds_.TransformBounds(transform);
+}
+
+bool RoundSuperellipseGeometry::CoversArea(const Matrix& transform,
+                                           const Rect& rect) const {
+  return false;
```

Comment:
I'd imagine computing a conservative inner rect for the rounded superellipse should be possible, no?